dsa/src/components.rs

```diff
@@ -81,7 +86,9 @@ impl<'a> DecodeValue<'a> for Components {
         let q = BigUint::from_bytes_be(q.as_bytes());
         let g = BigUint::from_bytes_be(g.as_bytes());
 
-        Ok(Self::from_components(p, q, g))
+        // TODO: Using this error doesn't seem right
```

Comment:
This is the correct error to use for the DER deserializer. You can go ahead and remove this comment.